src/components/ButtonDropDown/ButtonDropDown.js

```diff
@@ -0,0 +1,170 @@
+              }}
+            >
+              {items.map((item, index) => {
+                return (
+                  <li key={index}>
+                    <ButtonBase
+                      ref={node => (refs[index] = node)}
+                      onFocus={setHighlightedIndex(index)}
+                      onMouseOver={setHighlightedIndex(index)}
+                      css={`
+                       display: flex;
+                       align-items: center;
+                       height: 40px;
+                       width: calc(100% - 1px);
+                       border-left: 3px solid transparent;
+                       border-radius: 0;
+                       &:hover,
+                       &:focus {
+                         outline: none;
+                       }
+
+                       ${index === 0 &&
+                         `
+                           border-top-left-radius: 3px;
+                           border-top-right-radius: 3px;
+                         `}
+                       ${index === items.length - 1 &&
+                         `
+                           border-bottom-left-radius: 3px;
+                           border-bottom-right-radius: 3px;
+                         `}
+                       ${index === highlightedIndex &&
+                         `
+                           background: ${theme.surfaceHighlight};
+                           border-left: 3px solid ${theme.accent};
+                         `}
+                     `}
+                      onClick={handleItemClick(index)}
+                    >
+                      {item}
+                    </ButtonBase>
+                  </li>
+                )
+              })}
+            </List>
+          ))
+        /* eslint-enable react/prop-types */
+        }
+      </Transition>
+    </div>
+  )
+}
+
+ButtonDropDown.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  items: PropTypes.array,
+  cover: PropTypes.node,
+}
+
+const List = styled(animated.ul)`
+  position: absolute;
+  padding: 0;
+  margin: 0;
+  list-style: none;
+  border: ${({ border }) => `1px solid ${border}`};
+  border-radius: 4px;
+  background: ${({ surface }) => surface};
+  width: 100%;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.03);
+`
+
+function useButtonDropDown(onClick) {
+  const refs = []
+  const { highlightedIndex, setHighlightedIndex } = useArrowKeysFocus(refs)
+  const reset = useMemo(() => setHighlightedIndex(-1), [setHighlightedIndex])
+  const containerRef = useRef()
+  const [opened, setOpened] = useState(false)
+  const handleClose = useCallback(() => {
+    reset()
+    setOpened(false)
+  }, [reset, setOpened])
+  const handleItemClick = useCallback(
+    index => e => {
+      handleClose()
+      onClick(index)
+    },
+    [handleClose, onClick]
+  )
+  const { handleBlur } = useOnBlur(handleClose, containerRef)
+  useClickOutside(handleClose, containerRef)
```

Comment:
Nope `onBlur` works when another item receives focus and such item is not related to the container.

Reply:
What I mean is that when you click outside, whatever is focused in the container loses its focus, and the `blur` event should be triggered since React focus / blur events are bubbling?

Reply:
The way `onBlur` works (currently, we can update it) is it only checks for lost focus when another item received focus when there is no other item receiving focus it does not execute, maybe a better name would be `onLostFocus`, and potentially have a `onBlur` hook that integrates both `onClickoutside` and `onLostFocus`

Reply:
Oh right! I was assuming that `useOnBlur()` was adding an `onBlur` listener on the element passed to it.

What would you think of triggering the callback when `relatedTarget` is `null`? It seems like the correct behavior to me: we lost the focus of the current element and nothing else got focused, so the container and its children have effectively lost focus, which is what we want to know really.

And maybe we could name it differently yes, something like `useOnTreeBlur()` maybe?

Reply:
> What would you think of triggering the callback when relatedTarget is null?

I had a LOT OF FUN figuring out why the blur was happening in certain scenarios only to eventually find this is a way to separate both behaviours, that's why I suggested eventually to rename the hooks.

Reply:
Is there a need to separate both behaviors here? e.g. the Popover doesn’t https://github.com/aragon/aragon-ui/blob/master/src/components/Popover/Popover.js#L128-L141

Reply:
If it has already been implemented I can update once I can consume it from `aragonUI`

Reply:
Cool, no it’s not independent from Popover yet, I opened an issue => https://github.com/aragon/aragon-ui/issues/403

Reply:
That will be awesome and very handy :)